### PR TITLE
Pending talk claims

### PR DESCRIPTION
**Pull request by liam-wiltshire**
Create endpoint for pending talk claims

### CODE REVIEW

File: src/controllers/EventsController.php [hunk 1]
```diff
@@ -666,4 +684,33 @@ public function rejectAction($request, $db)
...
+    }
+
+    public function getPendingTalkClaimMapper($db, $request)
+    {
+        if (! isset($this->pending_talk_claim_mapper)) {
+            $this->pending_talk_claim_mapper = new PendingTalkClaimMapper($db, $request);
+        }
+
+        return $this->pending_talk_claim_mapper;
+    }
```
Comment (heiglandreas) [inline src/controllers/EventsController.php]: just an idea as I'm seeing a lot of setters and getters sprouting over the next commits: Can't we use Factory-Classes for getting the mappers? Either the default mapper is instantiated or you can call a static setter and then the set mapper will be returned. Basically moving the setMapper/getMapper into its own class.

Comment (heiglandreas) [inline src/controllers/EventsController.php]: And where is the test using it? 😛

File: src/models/PendingTalkClaimMapper.php [hunk 2]
```diff
@@ -167,4 +167,40 @@ public function approveClaimAsHost($talk_id, $speaker_id, $claim_id)
...
+                    'talk_uri'          => $base . '/' . $version . '/talks/' . $row['talk_id'],
+                    'speaker_uri'       => $base . '/' . $version . '/users/' . $row['speaker_id'],
+                    'approve_claim_uri' => ''
+                ];
+
+            }
+
+            return $retval;
+
+        } else {
```
Comment (heiglandreas) [inline src/models/PendingTalkClaimMapper.php]: This else can be omitted.

File: src/models/PendingTalkClaimMapper.php [hunk 3]
```diff
@@ -167,4 +167,40 @@ public function approveClaimAsHost($talk_id, $speaker_id, $claim_id)
...
+
+        $sql = 'select c.*, s.speaker_name from pending_talk_claims c
+                    inner join talks t on t.ID = c.talk_id
+                    inner join talk_speaker s on s.ID = c.claim_id
+                    where t.event_id = :event_id and
+                    (host_approved_at IS NULL or user_approved_at IS NULL)';
+        $stmt = $this->_db->prepare($sql);
+        $response = $stmt->execute(['event_id' => $event_id]);
+
+        if ($response && $stmt->rowCount() > 0) {
```
Comment (heiglandreas) [inline src/models/PendingTalkClaimMapper.php]: I'd alter this to
 ```php
if (! $response || $stmt->rowCount() < 1) {
    return false;
}
```
and then continue with the stuff you currently have in the else.

That way the "default flow" is going through linearily and exceptions fail fast

File: src/models/EventMapper.php [hunk 4]
```diff
@@ -453,6 +453,8 @@ public function transformResults($results, $verbose)
                             . $row['ID'] . '/comments/reported';
                         $list[$key]['reported_talk_comments_uri'] = $base . '/' . $version . '/events/'
                             . $row['ID'] . '/talk_comments/reported';
+                        $list[$key]['pending_claims_uri'] = $base . '/' . $version . '/events/'
```
Comment (liam-wiltshire) [inline src/models/EventMapper.php]: Erm... it is - check line 451 `if ($this->thisUserHasAdminOn($row['ID'])) {`

:-)

File: src/controllers/EventsController.php (hunk 1 above)
Comment (liam-wiltshire) [inline src/controllers/EventsController.php]: I agree, however I don't think it should form part of this PR. With ZendCon this week, I think we are well and truly in 'Conference Season', so I think getting this claim function live needs to be a priority.

That said, if we are going to aim to get test coverage up, we certainly need to do something, either DI or Factories - fancy raising a JIRA ticket? :-)

File: src/models/EventMapper.php (hunk 4 above)
Comment (heiglandreas) [inline src/models/EventMapper.php]: Ouch! That happens when you only review the changed code without having a look at the surrounding stuff....

Sorry 'bout that and forget my comment!

**Pull request merged**